include/circt/Dialect/HW/HWOpInterfaces.td

```diff
@@ -507,6 +507,14 @@ def HWInstanceLike : OpInterface<"HWInstanceLike", [
       "bool", "getDoNotPrint", (ins)
     >
   ];
+
+  let verify = [{
+    if (cast<circt::hw::HWInstanceLike>(op)
+            .getReferencedModuleNamesAttr()
+            .size())
+      return success();
+    return op->emitOpError("has empty list of referenced modules");
```

Comment:
I think you should use `$_op` instead of `op` directly.
I'd also fully qualify everything (i.e., also the `cast` and `success`)